objects/src/assets/token_symbol.rs

```diff
@@ -0,0 +1,114 @@
+use super::{AssetError, Felt, StarkField, ToString};
+use crate::utils::string::String;
+
+#[derive(Clone, Copy, Debug)]
+pub struct TokenSymbol(Felt);
+
+impl TokenSymbol {
+    pub const MAX_SYMBOL_LENGHT: usize = 6;
+    pub const MAX_ENCODED_VALUE: u64 = 26u64.pow(TokenSymbol::MAX_SYMBOL_LENGHT as u32);
+
+    pub fn new(symbol: &str) -> Result<Self, AssetError> {
+        let felt = encode_symbol_to_felt(symbol)?;
+        Ok(Self(felt))
+    }
+}
+
+impl From<TokenSymbol> for Felt {
+    fn from(symbol: TokenSymbol) -> Self {
+        symbol.0
+    }
+}
+
+impl TryFrom<&str> for TokenSymbol {
+    type Error = AssetError;
+
+    fn try_from(symbol: &str) -> Result<Self, Self::Error> {
+        TokenSymbol::new(symbol)
+    }
+}
+
+impl TryFrom<Felt> for TokenSymbol {
+    type Error = AssetError;
+
+    fn try_from(felt: Felt) -> Result<Self, Self::Error> {
+        // Check if the felt value is within the valid range
+        if felt.as_int() >= TokenSymbol::MAX_ENCODED_VALUE {
+            return Err(AssetError::TokenSymbolError("Encoded value is too large".to_string()));
+        }
+        let symbol = decode_felt_to_symbol(felt);
+        TokenSymbol::new(&symbol)
+    }
+}
+
+// HELPER FUNCTIONS
+// ================================================================================================
+// Utils to encode and decode the token symbol as a Felt. Token Symbols can consists of up to 6 characters
+// , e.g., A = 0, ...
+fn encode_symbol_to_felt(s: &str) -> Result<Felt, AssetError> {
+    if s.is_empty() || s.len() > TokenSymbol::MAX_SYMBOL_LENGHT {
+        return Err(AssetError::TokenSymbolError(
+            "Token Symbol must be of length >0 and <7".to_string(),
```

Comment:
minor nit: I'd change this to: "Token symbol must be between 1 and 6 characters long."